src/libs/antares/study/area/links.cpp

```diff
@@ -72,6 +72,25 @@ AreaLink::~AreaLink()
 {
 }
 
+// Forward declarations
+static bool linkLoadTimeSeries_for_version_under_320(AreaLink& link, const AnyString& folder, Study& study);
+static bool linkLoadTimeSeries_for_version_from_320_to_630(AreaLink& link, const AnyString& folder);
+static bool linkLoadTimeSeries_for_version_from_630_to_810(AreaLink& link, const AnyString& folder);
+static bool linkLoadTimeSeries_for_version_820_and_later(AreaLink& link, const AnyString& folder);
+
+bool AreaLink::loadTimeSeries(Study& study, const AnyString& folder)
+{
+    if (study.header.version < 320)
+        return linkLoadTimeSeries_for_version_under_320(*this, folder, study);
+    else if (study.header.version < 630)
+        return linkLoadTimeSeries_for_version_from_320_to_630(*this, folder);
+    else if (study.header.version < 820)
+        return linkLoadTimeSeries_for_version_from_630_to_810(*this, folder);
+    else {
+        return linkLoadTimeSeries_for_version_820_and_later(*this, folder);
```

Comment:
Since you pass `*this`, it seems natural to make all these functions private members of class `AreaLink`.

Reply:
ok, done